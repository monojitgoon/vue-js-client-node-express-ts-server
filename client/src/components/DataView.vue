<template>
  <div>
    <div v-for="(row, rindex) in offersData" :key="rindex" class="data_row">
      <div class="item">
        <div class="item__column">
          <div class="item__card">
            <h4>Departure</h4>
            <i class="fas fa-plane-departure"></i>
            <p>Date: {{ row.departureDate }}</p>
            <p>Origin: {{ row.origin }}</p>
          </div>
        </div>
        <div class="item__column">
          <div class="item__card">
            <h4>Offer</h4>
            <i class="fas fa-tags"></i>
            <p>Price: {{ row.priceAmount }} {{ row.priceCurrency }}</p>
            <p>Seat Available: {{ row.seatAvailability }}</p>
          </div>
        </div>
        <div class="item__column">
          <div class="item__card">
            <h4>Arrival</h4>
            <i class="fas fa-plane-arrival"></i>
            <p>Date: {{ row.returnDate }}</p>
            <p>Destination: {{ row.destination }}</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'DataView',
  props: {
    offersData: { type: Array, default: () => [] },
  },
  data() {
    return {};
  },
  computed: {
    isEmpty: function(value) {
      return Object.keys(value).length === 0 && value.constructor === Object;
    },
  },
};
</script>

<style scoped lang="scss">
@import '../styles/base/general';

.data_row {
  width: 100%;
  padding: 20px;
  box-shadow: $shadow-small;
  display: flex;
  margin: 20px 0;
  position: relative;
}

.item {
  width: 100%;
  line-height: 1.5;
}

.item:after {
  content: '';
  display: table;
  clear: both;
}

.item__column {
  float: left;
  width: 33%;
  padding: 0 10px;

  @include mobile {
    width: 100%;
    display: block;
    margin-bottom: 20px;
  }

  @include tablet {
    width: 100%;
    display: block;
    margin-bottom: 20px;
  }
}
.item__card {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  padding: 16px;
  text-align: center;
  background-color: #f1f1f1;
}

.item__card > i {
  margin-right: 10px;
}

.item__card > p {
  margin: 0;
}
</style>
