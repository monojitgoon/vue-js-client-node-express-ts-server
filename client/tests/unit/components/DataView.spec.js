import { shallowMount } from '@vue/test-utils';
import DataView from '@/components/DataView';

describe('DataView', () => {
  it('renders props.offersData when passed', async () => {
    const offersData = [
      {
        origin: 'CBB',
        destination: 'DDD',
        departureDate: '2021-08-03',
        returnDate: '2022-08-01',
        seatAvailability: 34,
        priceAmount: 995.72,
        priceCurrency: 'EUR',
      },
    ];
    const wrapper = shallowMount(DataView, {
      propsData: { offersData: offersData },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
