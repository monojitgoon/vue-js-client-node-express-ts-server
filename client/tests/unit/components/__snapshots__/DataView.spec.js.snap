// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataView renders props.offersData when passed 1`] = `
<div>
  <div class="data_row">
    <div class="item">
      <div class="item__column">
        <div class="item__card">
          <h4>Departure</h4> <i class="fas fa-plane-departure"></i>
          <p>Date: 2021-08-03</p>
          <p>Origin: CBB</p>
        </div>
      </div>
      <div class="item__column">
        <div class="item__card">
          <h4>Offer</h4> <i class="fas fa-tags"></i>
          <p>Price: 995.72 EUR</p>
          <p>Seat Available: 34</p>
        </div>
      </div>
      <div class="item__column">
        <div class="item__card">
          <h4>Arrival</h4> <i class="fas fa-plane-arrival"></i>
          <p>Date: 2022-08-01</p>
          <p>Destination: DDD</p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
